### ISSUE TITLE
issue with one-class segmentation tutorial 

### ISSUE THREAD

**Issue by clem-L** [ivadomed/ivadomed#448]
Hi,

I tried to do the one-class segmentation tutorial.
I don't think I got execution errors on the train or eval part but Dice score are not as good as in the tutorial. 
Pred_mask take some voxels far from the spinal cord.

What should I change to get same results as in the tutorial ? 

`Loading model: spineGeneric/best_model.pt
Inference - Iteration 0: 100%|████████████████████| 5/5 [00:08<00:00,  1.80s/it]
{'dice_score': 0.12947206694882588, 'multi_class_dice_score': 0.7225174405080848, 'hausdorff_score': 0.2163740187077797, 'precision_score': 0.12528217864784708, 'recall_score': 0.13395235268724368, 'specificity_score': 0.9994632195317591, 'intersection_over_union': 0.07869927791501147, 'accuracy_score': 0.9984023848906294}

Run Evaluation on spineGeneric/pred_masks

Evaluation: 100%|█████████████████████████████████| 5/5 [00:48<00:00,  9.63s/it]
                         avd_class0  ...  vol_pred_class0
image_id                             ...                 
sub-strasbourg04_T1w       0.658537  ...          1680.00
sub-strasbourg04_T2w       0.051829  ...          5175.00
sub-strasbourg04_T2star    0.998841  ...             5.00
sub-hamburg01_T1w          0.657743  ...          1933.75
sub-hamburg01_T2w          0.059071  ...          5983.75

[5 rows x 17 columns]
                         avd_class0  ...  vol_pred_class0
image_id                             ...                 
sub-strasbourg04_T1w       0.658537  ...          1680.00
sub-strasbourg04_T2w       0.051829  ...          5175.00
sub-strasbourg04_T2star    0.998841  ...             5.00
sub-hamburg01_T1w          0.657743  ...          1933.75
sub-hamburg01_T2w          0.059071  ...          5983.75

[5 rows x 17 columns]
`

![seg_manual_tutorial](https://user-images.githubusercontent.com/71508966/93607014-12f4d780-f9c9-11ea-9de1-b399b4580345.png)

**Comment by andreanne-lemay** [comment]
Hi @clem-L , thank you for opening this issue! Can you share the configuration file you used to train the model? It might help us to figure out what to change in order to have a better performance.

**Comment by clem-L** [comment]
Here the configuration file that I used: 

{
    "command": "train",
    "gpu": 0,
    "log_directory": "spineGeneric",
    "model_name": "seg_sc_t1_t2_t2s_mt",
    "debugging": false,
    "object_detection_params": {
        "object_detection_path": null
    },
    "loader_parameters": {
        "bids_path": "/mnt/software/sct_4.3/data/data_spinegeneric_registered",
        "target_suffix": ["_seg-manual"],
        "roi_params": {
            "suffix": null,
            "slice_filter_roi": null
        },
        "contrast_params": {
            "training_validation": ["T1w", "T2w", "T2star"],
            "testing": ["T1w", "T2w", "T2star"],
            "balance": {}
        },
        "slice_filter_params": {
            "filter_empty_mask": false,
            "filter_empty_input": true
        },
        "slice_axis": "axial",
        "multichannel": false,
        "soft_gt": false
    },
    "split_dataset": {
        "fname_split": null,
        "random_seed": 6,
        "center_test": [],
        "method": "per_patient",
        "train_fraction": 0.6,
        "test_fraction": 0.2
    },
    "training_parameters": {
        "batch_size": 18,
        "loss": {
            "name": "DiceLoss"
        },
        "training_time": {
            "num_epochs": 100,
            "early_stopping_patience": 50,
            "early_stopping_epsilon": 0.001
        },
        "scheduler": {
            "initial_lr": 0.001,
            "lr_scheduler": {
                "name": "CosineAnnealingLR",
                "base_lr": 1e-5,
                "max_lr": 1e-2
            }
        },
        "balance_samples": false,
        "mixup_alpha": null,
        "transfer_learning": {
            "retrain_model": null,
            "retrain_fraction": 1.0
        }
    },
    "default_model": {
        "name": "Unet",
        "dropout_rate": 0.3,
        "bn_momentum": 0.9,
        "depth": 4
    },
    "FiLMedUnet": {
        "applied": false,
        "metadata": "contrasts",
        "film_layers": [0, 1, 0, 0, 0, 0, 0, 0, 0, 0]
    },
    "testing_parameters": {
        "binarize_prediction": true,
        "uncertainty": {
            "epistemic": false,
            "aleatoric": false,
            "n_it": 0
        }
    },
    "evaluation_parameters": {
        "targetSize": {"unit": "vox", "thr": [20, 100]},
        "removeSmall": {"unit": "vox", "thr": 3},
        "overlap": {"unit": "vox", "thr": 3}
    },
    "transformation": {
        "Resample":
        {
            "wspace": 0.75,
            "hspace": 0.75,
            "dspace": 1,
            "preprocessing": true
        },
        "CenterCrop": {
            "size": [128, 128],
            "preprocessing": true
        },
        "RandomAffine": {
            "degrees": 5,
            "scale": [0.1, 0.1],
            "translate": [0.03, 0.03],
            "applied_to": ["im", "gt"],
            "dataset_type": ["training"]
        },
        "ElasticTransform": {
			"alpha_range": [28.0, 30.0],
			"sigma_range":  [3.5, 4.5],
			"p": 0.1,
            "applied_to": ["im", "gt"],
            "dataset_type": ["training"]
        },
      "NumpyToTensor": {},
      "NormalizeInstance": {"applied_to": ["im"]}
    }
}

**Comment by andreanne-lemay** [comment]
@clem-L Everything looks good to me in the configuration file. How was the loss progression during the training? If you don't have the prints in your terminal, you can visualize the training on tensorboard by executing this command in the location where you ran the training: `tensorboard --logdir spineGeneric` (spineGeneric being the name of your log directory). You can then access in your browser to `localhost:6006` and you should get a page looking like this:
![tensorboard_scalar](https://user-images.githubusercontent.com/49137243/93625235-9d304200-f9af-11ea-9f12-f6a5a093aafd.png)

In the tab "Images", you should also be able to visualize the model's prediction.

If possible, can you include screenshots of the validation metrics from tensorboard and the image predictions? This will help to know if the problem occurred during the training or the evaluation.

**Comment by clem-L** [comment]
Hi, 
Here I include screenshot of the validation metrics but I can't see the image predictions.
I am not sure but the loss progression seems OK ?
 
![tensorboard](https://user-images.githubusercontent.com/71508966/93755784-2cce2e80-fc04-11ea-9ae8-040cc833a375.png)

**Comment by charleygros** [comment]
Hi @clem-L ,

Thanks for sharing your TensorBoard! Looks good to me as well

May I ask you which version of `ivadomed` are you using?
Are you using the `ivadomed` installed with SCT? cf [here](https://github.com/neuropoly/spinalcordtoolbox/blob/master/requirements.txt#L5).
If so, may I ask you to update the `ivadomed` version to the latest version (v2.1.0 or `master`)? see documentation [here](https://ivadomed.org/en/latest/installation.html#installation). And try again.
Cheers, Charley

**Comment by clem-L** [comment]
Hi,

I work under a virtual environment and installed ivadomed with pip.
I tried to run the inference again and got voxels far from the spinal cord.

![new_tensorboard](https://user-images.githubusercontent.com/71508966/93865671-eab9f100-fcc6-11ea-86f3-33182cdbd4e0.png)

![new_segmentation_2](https://user-images.githubusercontent.com/71508966/93865685-f0173b80-fcc6-11ea-86b8-4091d323e147.png)

**Comment by charleygros** [comment]
Hi @clem-L ,

Many thanks for your answer.

May I ask you to run the following command and to copy and paste the terminal output here:
```
pip show ivadomed
```

I am scared that you are not using the last version of `ivadomed`. For instance in the config file you posted [here](https://github.com/ivadomed/ivadomed/issues/448#issuecomment-694917382), the parameter `binarize_prediction` has the value `true`, which is not the case in the last version of ivadomed, see [here](https://github.com/ivadomed/ivadomed/blob/master/ivadomed/config/config.json#L76).
By updating your `ivadomed` I am confident that you should get similar results than the one showed in the tutorial. Please let us know if the problem remains, we would be happy to help.

Cheers,
Charley

**Comment by olix86** [comment]
Hi @clem-L 

We just released a new version (2.2.1) which should fix your issue, you can upgrade with `pip install ivadomed --upgrade`. 

Let us know if the issue persists,

Olivier

**Comment by clem-L** [comment]
Hi, 

I upgrade to the new ivadomed version and I got same results as in the tutorial.
